src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTestBase.cs

```diff
@@ -55,13 +59,179 @@ public async Task<PullRequest> WaitForPullRequestAsync(string targetRepo, string
             throw new MaestroTestException($"No pull request was created in {targetRepo} targeting {targetBranch}");
         }
 
+        public async Task<PullRequest> WaitForUpdatedPullRequestAsync(string targetRepo, string targetBranch, int attempts = 7)
+        {
+            Repository repo = await GitHubApi.Repository.Get(_parameters.GitHubTestOrg, targetRepo).ConfigureAwait(false);
+            PullRequest pr = await WaitForPullRequestAsync(targetRepo, targetBranch);
+
+            while (attempts-- > 0)
+            {
+                pr = await GitHubApi.PullRequest.Get(repo.Id, pr.Number);
+
+                if (pr.CreatedAt != pr.UpdatedAt)
+                {
+                    return pr;
+                }
+
+                await Task.Delay(60 * 1000).ConfigureAwait(false);
+            }
+
+            throw new MaestroTestException($"The created pull request for {targetRepo} targeting {targetBranch} was not updated with subsequent subscriptions after creation");
+        }
+
+        public async Task<bool> WaitForMergedPullRequestAsync(string targetRepo, string targetBranch, int attempts = 7)
+        {
+            Repository repo = await GitHubApi.Repository.Get(_parameters.GitHubTestOrg, targetRepo).ConfigureAwait(false);
+            PullRequest pr = await WaitForPullRequestAsync(targetRepo, targetBranch);
+
+            while (attempts-- > 0)
+            {
+                TestContext.WriteLine($"Starting check for merge, attempts remaining {attempts}");
+                pr = await GitHubApi.PullRequest.Get(repo.Id, pr.Number);
+
+                if (pr.State == ItemState.Closed)
+                {
+                    return true;
+                }
+
+                await Task.Delay(60 * 1000).ConfigureAwait(false);
+            }
+
+            throw new MaestroTestException($"The created pull request for {targetRepo} targeting {targetBranch} was not merged within {attempts} minutes");
+        }
+
+        private async Task<Microsoft.DotNet.DarcLib.PullRequest> WaitForAzDoPullRequestAsync(string targetRepoUri, string targetBranch)
+        {
+            int attempts = 10;
+            while (attempts-- > 0)
+            {
+                var prs = await AzDoClient.SearchPullRequestsAsync(targetRepoUri, targetBranch, Microsoft.DotNet.DarcLib.PrStatus.Open).ConfigureAwait(false);
```

Comment:
I'd just make the changes so that this works as part of this PR.

Reply:
I think I'll actually go the other direction and pull them out, since the code isn't referenced anywhere yet (this must have gotten pulled in from my other branch).